test/AutoRest.TestServer.Tests/Infrastructure/TestServerV1.cs

```diff
@@ -15,40 +15,35 @@ namespace AutoRest.TestServer.Tests.Infrastructure
 {
     public class TestServerV1 : IDisposable, ITestServer
     {
-        private static Regex _scenariosRegex = new Regex("(coverage|optionalCoverage|optCoverage)\\[(\"|')(?<name>\\w+)(\"|')\\]", RegexOptions.Compiled);
+        private static readonly Regex _scenariosRegex = new Regex("(coverage|optionalCoverage|optCoverage)\\[(\"|')(?<name>\\w+)(\"|')\\]", RegexOptions.Compiled);
 
-        private Process _process;
+        private readonly Process _process;
         public HttpClient Client { get; }
         public string Host { get; }
 
         public TestServerV1()
         {
             var portPhrase = "Server started at port ";
             var startup = Path.Combine(GetBaseDirectory(), "legacy", "startup", "www.js");
-
-            var processStartInfo = new ProcessStartInfo("node", startup);
-
-            // Use random port
-            processStartInfo.Environment["PORT"] = "0";
-            processStartInfo.RedirectStandardOutput = true;
-            processStartInfo.RedirectStandardError = true;
+            var port = new Random().Next(10000, 65536).ToString();
```

Comment:
For nextLink to provide the correct port on the test server, the port must be placed into the "PORT" environment variable. This cannot be done after the process has started, so we randomize the port before starting the process. These changes should work for both V1 and V2. However, there was a V2 issue in which it wouldn't use the environment port value properly. Laurent made an [update to the test server](https://github.com/Azure/autorest.testserver/pull/120) that should allow the tests to run on V2, but this PR doesn't update the test server. So, we can try it in a subsequent PR.

Reply:
This makes test server startup flaky. The random port we generate might be taken

Reply:
https://github.com/Azure/autorest.testserver/pull/121